posts/python-in-production-part-3-of-5.md

```diff
@@ -0,0 +1,107 @@
+* `--compressed` tells shiv to compress the files for the smallest possible binary.
+* `-o simple_api` tells shiv what to name our executable.
+* `-e simple_api.__main__:main src` tells shiv the entry point to our program and where to go looking for it.
+* `--upgrade` tells shiv to replace our binary if this is a subsequent run.
+
+### Keep your Requirements Files Separate
+
+Keeping your requirements files for pip separate makes it easy to setup your environment for development mode vs. building.  This is especially true when cloning this project onto a new machine.  The steps in this series are to aid in the development and release of large python projects, we use a simple example here to give you the tools to build off of in the future.  
+
+### Keep your Virtual Environments Separate
+
+Having multiple virtual environments helps to keep things clean.  Notice how in the build script we activate the virtual environment that lives in the dist folder, but we get the site packages for our binary from our development virtual environment.  That is because the dependencies required to run shiv are different then the ones required to build our program.
+
+### A Note About Building Remotely
+
+In a subsequent part of this series, we will put this process into a Jenkins build server.  When that happens, it will be necessary to create a separate bash script for setting up the initial virtual environment (refered to as the development virtual environment above) and installing our programs dependencies, because it will not exist on a newly provisioned machine.
+
+# Running the Executable Binary
+
+Congratulations! You have just created a single file, binary executable.  In order to run the binary, we must go back to the terminal.
+```bash
+./simple_api
+```
+I encourage you to put this executable binary onto a different machine and run it!  As long as that machine has the proper version of Python installed, the program will run.  
+
+### CPU Architecture Matters
+
+There are a few things worth mentioning that could cause you some errors along the way, especially if your program is not *pure python*.  If you build this binary on a machine running an ARM processor, and then copy it to a machine running an AMD processor, you may run into some issues.  In a later part of the series, we will discuss building for different architectures in more depth.  Remember from part 1, I mentioned that it is good to know which version of python you are building for?  Well, it is also good to know which architecture you are building for too!  
+
+Some packages that are architecture specific may include things like Audio Controllers, Systemd bindings, or SPI dev kits.  Mainly because these packages interact with the hardware of the computer and therefore are more architecture specific.
```

Comment:
Update `Mainly because these packages interact with the hardware of the computer and therefore are more architecture specific.` to `These packages interact with the hardware of the computer and therefore are more architecture-specific.`